app/components/certificate/certificate-request.tsx

```diff
@@ -0,0 +1,51 @@
+import { Flex, Heading, Text, HStack, Button } from '@chakra-ui/react';
+
+import Description from './description';
+
+interface CertificateRequestProps {
+  subject: string;
+  validFrom: Date;
+  validTo: Date;
+  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
```

Comment:
These seem wrong.  Take an `onClick` in the props, and move the `handleSubmit` function up to the parent component.